app/models/list_import/anilist.rb

```diff
@@ -36,77 +12,122 @@ class Anilist < ListImport
     validate :ensure_user_exists, on: :create
 
     def ensure_user_exists
-      return if input_text.blank?
-
-      @auth_token ||= get_auth_token
-      request = Typhoeus::Request.get(build_url(input_text))
+      return false if input_text.blank?
+      return true if user_exists?
 
-      case request.code
-      when 404
-        errors.add(:input_text, 'Anilist user not found.')
-      end
+      errors.add(:input_text, "AniList user not found - #{input_text}")
     end
 
     def count
-      # animelist will get me everything
-      get("#{input_text}/animelist")['stats']['status_distribution'].map { |type|
-        # if user does not have any stats return 0
-        next 0 if type.last.count.zero?
-
-        type.last.values.inject(&:+)
-      }.inject(&:+)
+      @count ||= anime_list.count + manga_list.count
     end
 
     def each
-      # pass in toyhammered/#{anime}list
       %w[anime manga].each do |type|
-        list = get("#{input_text}/#{type}list")['lists']
-        # in case there are no anime/manga
-        # need to prevent an error being raised
-        list&.each do |status|
-          # status[0] is always going to be either
-          # completed, on-hold, etc... just skip over that part
-          status[1].each do |media|
-            row = Row.new(media, type)
+        send("#{type}_list").each do |media|
+          row = Row.new(media, type)
 
-            yield row.media, row.data
-          end
+          yield row.media_mapping, row.data
```

Comment:
Shouldn't this be row.media?

Reply:
I actually changed to media_mapping thinking it was more understandable but now I am questioning that :P

Reply:
I have changed back to be row.media though.